### PR TITLE
Refactor image list-catalog data source and documentation

### PR DESCRIPTION
**Pull request by Diptipowervs**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

### CODE REVIEW

**Review by michaelkad — COMMENTED** [review]
LGTM

**Review by ismirlia — DISMISSED** [review]

**Comment by ismirlia** [comment]
I think this looks good, if @michaelkad approves, open the PR in the main terraform repo for merging there.

File: ibm/service/power/ibm_pi_constants.go [hunk 1]
```diff
@@ -47,27 +282,34 @@ const (
...
-	PIInstanceDeploymentType             = "pi_deployment_type"
-	PIInstanceNetwork                    = "pi_network"
-	PIInstanceStoragePool                = "pi_storage_pool"
-	PISAPInstanceProfileID               = "pi_sap_profile_id"
-	PISAPInstanceDeploymentType          = "pi_sap_deployment_type"
-	PIInstanceStoragePoolAffinity        = "pi_storage_pool_affinity"
-	Arg_PIInstanceSharedProcessorPool    = "pi_shared_processor_pool"
+	CUSTOM_VIRTUAL_CORES = "custom-virtualcores"
+
+	//Arg_CloudInstanceID                 = "pi_cloud_instance_id"
```
Comment (michaelkad) [inline ibm/service/power/ibm_pi_constants.go:294]: This is in line 8 and should be removed

**Review by michaelkad — CHANGES_REQUESTED** [review]
Just one thing

**Review by ismirlia — APPROVED** [review]

**Review by michaelkad — APPROVED** [review]

**Comment by ismirlia** [comment]
@Diptipowervs Since @michaelkad approved now could you open this pull request in the main terraform provider repo to get this ready to merge?

**Comment by Diptipowervs** [comment]
PR opened in main

**Pull request closed without merging**